### PR TITLE
Trim ids collected and remove remaining bytes at the end of them

### PR DESCRIPTION
**Pull request by dinosaure**
PR checklist:

- [x] Purpose of the code is [evident to future readers](https://semgrep.dev/docs/contributing/contributing-code/#explaining-code)
- [x] Tests included or PR comment includes a reproducible test plan
- [x] Documentation is up-to-date
- [ ] A changelog entry was [added to changelog.d](https://semgrep.dev/docs/contributing/contributing-code/#adding-a-changelog-entry) for any user-facing change
- [x] Change has no security implications (otherwise, ping security team)

If you're unsure about any of this, please see:

- [Contribution guidelines](https://semgrep.dev/docs/contributing/contributing-code)!
- [One of the more specific guides located here](https://semgrep.dev/docs/contributing/contributing/)

This PR trims `ids` collected by `Nosem.ml` and remove remaining bytes that are not a part of the `id`.

### CODE REVIEW

**Review by aryx — APPROVED** [review]

File: src/reporting/Nosemgrep.ml [hunk 1]
```diff
@@ -124,6 +135,13 @@ let rule_match_nosem ~strict (rule_match : Out.cli_match) :
           (Option.value ~default:[||] ids_previous_line)
       in
       let ids = Common.map_filter Fun.id (Array.to_list ids) in
+      let ids = Common.map String_utils.to_unicode_string ids in
+      let ids = Common.map Result.to_option ids in
+      let ids = Common.map_filter Fun.id ids in
+      let ids = Common.map (String_utils.unicode_span ~sat:no_space) ids in
+      let ids = Common.map fst ids in
+      let ids = Common.map String_utils.unicode_string_to_utf_8_string ids in
+      Fmt.epr "%a\n%!" Fmt.(Dump.list string) ids;
```
Comment (aryx) [inline src/reporting/Nosemgrep.ml]: Also probably better to remove those debugging instructions.

**Review by aryx — CHANGES_REQUESTED** [review]
Comments, deforest, and probably update the dune and dune-project files are required.

**Comment by hannesm** [comment]
maybe a stupid question, but the `span` and unicode stuff seems to be like a big hammer... isn't it sufficient to get the rule_id as input, and use `String.index` with the space, tab, newline character -- and use the one that has the lowest index thereof for String.sub (or Str.first_chars)? when I understand this correctly, a rule id is terminated by any whitespace character (certainly, a question would be what to do with unicode whitespace characters, or non-printable characters (such as BEL))...

**Comment by aryx** [comment]
yep, I agree with hannes, I wonder if there's a simpler way to solve it.

**Comment by dinosaure** [comment]
I updated the PR according to your review and simplify a bit the process :+1:.

**Pull request merged**